src/server_pool.rs

```diff
@@ -1,16 +1,13 @@
 use crate::Server;
 use crate::{Error, ErrorKind};
-use lazy_static::lazy_static;
 use std::collections::VecDeque;
 use std::ops::{Deref, DerefMut, Drop};
-use std::sync::{Arc, Mutex};
+use std::sync::Mutex;
 use tokio::sync::{Semaphore, SemaphorePermit};
 
-const DEFAULT_POOL_SIZE: usize = 50;
-
-lazy_static! {
-    pub(crate) static ref SERVER_POOL: ServerPool = ServerPool::new(DEFAULT_POOL_SIZE);
-}
+// macOS has small default ulimits
+const DEFAULT_POOL_SIZE: usize = if cfg!(target_os = "macos") { 20 } else { 50 };
```

Comment:
These tests happen to pass currently, because they have a bug — the servers vec is emptied on every iteration, so they never exhaust the pool.

But when these tests actually try to get the whole pool, they cause a deadlock whenever any other test runs in parallel that uses more than one server at a time.

Reply:
right, that vec was supposed to be outside the for loop :facepalm: